go/adbc/driver/flightsql/flightsql_connection.go

```diff
@@ -225,6 +225,7 @@ func (c *connectionImpl) getSessionOptions(ctx context.Context) (map[string]inte
 
 func (c *connectionImpl) setSessionOptions(ctx context.Context, key string, val interface{}) error {
 	req := flight.SetSessionOptionsRequest{}
+	ctx = metadata.NewOutgoingContext(ctx, c.hdrs)
```

Comment:
should we use `AppendToOutgoingContext` instead? Just in case there's already an existing context so that we're appending the headers rather than dropping any existing ones?